### ISSUE TITLE
Procon crashing after SQL error

### ISSUE THREAD

**Issue by Grossincome** [AdKats/AdKats#908]
I get the following message after running fine for a while. I had same error on laptop so create a fresh instance on amazon instance. Works fine for a while and error and crashes.

[10:30:23 95] [Statslogger]Error: Error in UpdateCurrentPlayerTable: 
[10:30:23 96] Message: Incorrect string value: '' for function inet_aton
[10:30:23 96] Native: -2147467259
[10:30:23 96] Source: MySql.Data
[10:30:23 96] StackTrace:    at MySql.Data.MySqlClient.MySqlStream.ReadPacket()
   at MySql.Data.MySqlClient.NativeDriver.GetResult(Int32& affectedRow, Int64& insertedId)
   at MySql.Data.MySqlClient.Driver.GetResult(Int32 statementId, Int32& affectedRows, Int64& insertedId)
   at MySql.Data.MySqlClient.Driver.NextResult(Int32 statementId, Boolean force)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResult()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReader(CommandBehavior behavior)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at PRoConEvents.CChatGUIDStatsLogger.UpdateCurrentPlayerTable(List`1 lstPlayers)
[10:31:24 35] [Statslogger]Error: Error in UpdateCurrentPlayerTable: 
[10:31:24 37] Message: Incorrect string value: '' for function inet_aton
[10:31:24 37] Native: -2147467259
[10:31:24 37] Source: MySql.Data
[10:31:24 37] StackTrace:    at MySql.Data.MySqlClient.MySqlStream.ReadPacket()
   at MySql.Data.MySqlClient.NativeDriver.GetResult(Int32& affectedRow, Int64& insertedId)
   at MySql.Data.MySqlClient.Driver.GetResult(Int32 statementId, Int32& affectedRows, Int64& insertedId)
   at MySql.Data.MySqlClient.Driver.NextResult(Int32 statementId, Boolean force)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResult()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReader(CommandBehavior behavior)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at PRoConEvents.CChatGUIDStatsLogger.UpdateCurrentPlayerTable(List`1 lstPlayers)

**Comment by danfango** [comment]
Hi i had same problem it I think its the CChatGUIDStatsLogger plugin try this one instead it seems to work better :

CChatGUIDStatsLogger-1.0.0.2-ty_gerMOD-master

**Comment by ColColonCleaner** [comment]
This happened for other people but it should be fixed in the latest test version.

#872 

Basically AdKats was putting blank IP addresses into the database instead of NULL, which stat logger doesn't have handling for. It just assumes and runs with it.

**Comment by danfango** [comment]
I'm sure I had the latest version of both plugins and my procon was crashing (first crash I ever saw in procon) after the original statslogger was removed no crashing put it back in crashed. I found the modmaster version by chance it seems stable no crashes in procon but I still have to check how well it actually works with adkats

**Comment by ColColonCleaner** [comment]
Well yes, the error is thrown by stat logger, but the cause of the error was AdKats. That exception should not normally cause procon to crash, but on some people's systems it does.

Download the *.112 version that i linked you and test that out for a while.

**Comment by Semperfi-mantis** [comment]
hello guys. im starting with procon and mysql.. i followed lost of tutorials on how to set it up. im using a localhost connection on one pc that runs procon and mysql.. and all is working good, accept procon..  on a empty server it runs fine(with me in the server and me with a other account on other pc) but when i do the same on a half/full server procon crashes i get this error (same as above). and i cant figure it out how to fix it.. Thanks in advance!

[Statslogger]Error: Error in UpdateCurrentPlayerTable: 
[23:16:33 56] Message: Incorrect string value: '''' for function inet_aton
[23:16:33 57] Native: -2147467259
[23:16:33 58] Source: MySql.Data
[23:16:33 59] StackTrace:    bij MySql.Data.MySqlClient.MySqlStream.ReadPacket()
   bij MySql.Data.MySqlClient.NativeDriver.GetResult(Int32& affectedRow, Int64& insertedId)
   bij MySql.Data.MySqlClient.Driver.GetResult(Int32 statementId, Int32& affectedRows, Int64& insertedId)
   bij MySql.Data.MySqlClient.Driver.NextResult(Int32 statementId, Boolean force)
   bij MySql.Data.MySqlClient.MySqlDataReader.NextResult()
   bij MySql.Data.MySqlClient.MySqlCommand.ExecuteReader(CommandBehavior behavior)
   bij MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   bij PRoConEvents.CChatGUIDStatsLogger.UpdateCurrentPlayerTable(List`1 lstPlayers)

**Comment by danfango** [comment]
First off Procon is complete shit .... it's free software written by incompetent australian idiots and to compound the disaster there are even bigger idiots writing really crap plugins. The crash you are referring to is probably the PRoCon Chat, GUID, Stats and Map Logger - 1.0.0.2 crash which even with a clean procon crashes it and if you use the updated plugin it's almost as bad. Adkats while a good plugin relies on totally crap second party plugins my advice delete procon and move on to another piece of better software or give up renting your server cause procon rconnet and BF4DB/Metabans only lines the pockets of incompetent untested wanna be's who could not write their own f'in name let a lone decent software

**Comment by danfango** [comment]
P.S. fuck you coloncleaner you are a fucking dick that exploits normal players get a fukcing life

**Comment by danfango** [comment]
procon is shit and so are your fucking plugins no wonder they are fucking free

**Comment by danfango** [comment]
@  Semperfi-mantis ps. you have no fucking chance I have 37 years in I.T.  and I could not get the fucking software to work (procon, adkats, stats logger, mysql). It's flawed it's crap garbage in garbage out

**Comment by ColColonCleaner** [comment]
@danfango I see you have some very strong opinions. I remember your comments back in #910 where you were having stability problems. However, considering even now with the decline of BF4/BF3 stat logger and AdKats are still actively running on 1000+ servers I would take your implementation into question.

AdKats was started as a basic admin project a couple years before I finished college and continued on as hobby. Even after 4500 hours of work on it the plugin is still free because I find it enjoyable to contribute to the community. The only paid portions are specific features people want implemented their way.

I'm wondering where you got such a negative opinion of me from. It's very odd. Could you elaborate?

**Comment by ColColonCleaner** [comment]
@Prophet731 I remember seeing this inet_aton issue previously. Do you remember what went on with that?

**Comment by danfango** [comment]
Good luck if they are running on 1000+ servers must be luck cause the stats logger plugin (both versions) is complete crap and crashes procon even on a clean install

**Comment by ColColonCleaner** [comment]
@Semperfi-mantis I remember seeing the issue you mentioned previously, but I'll need to look back over my previous patches to remember what we did.

**Comment by danfango** [comment]
I doubt those 100 + server have 37 years in I.T. and are grossing 35 k

**Comment by danfango** [comment]
Si9mplr the stats logger CRASHES PROCON even a n idiot can see that

**Comment by danfango** [comment]
He even commented that it's stable till somene enters the server DOH

**Comment by danfango** [comment]
thats the EXACT crash

**Comment by danfango** [comment]
write a decent plugin or don#t fuckign bother really yu guys need to get out more

**Comment by Semperfi-mantis** [comment]
@danfango why so negative.. be glad there a people out there who make this.. 

i know that its stats logger who causes this crashing of procon.. its not adkats its self, adkats is running just fine, even now and i have stats logger plugin off.. ill wait on ColColonCleaner's responds.. 

if you have 37 years I.T why aren't you @danfango helping them or even making such things for the community..  dont be so negative mate

**Comment by danfango** [comment]
oh wait i be glad they make software that crashes really ?

**Comment by danfango** [comment]
I don;t dabble in procon or plugins cause I have no expertise there seems others are not so caring

**Comment by danfango** [comment]
I paid coloncleaner to stop spamming my server with messages lol really !

**Comment by ColColonCleaner** [comment]
@danfango I've set up dozens of different groups with procon and AdKats personally without issues. At its height AdKats was running on 3000 servers without people mentioning these crashes.

It appears you and at least one other person are having a certain issue. Extrapolating that out to the entire population is short-sighted, and not evident of a person who has experience with software.

**Comment by danfango** [comment]
anyway I'm leaving BF4 too many idiots on and off the field and it's costing me way too much money, time and effort

**Comment by danfango** [comment]
lucky them i've tested it on many platforms and installs and really it's flawed maybe they are running windows xp ?

**Comment by danfango** [comment]
how can a plugin crash procon in a minute but when you switch it off procon os stable ?

**Comment by danfango** [comment]
im fuckign 56 years old dude I was here before probably you were born have some respect and don't throw 100 users are ok that means nothing

**Comment by danfango** [comment]
and @ColColonCleaner why did you charge me to switch off your spamming shit in my server  ?

**Comment by danfango** [comment]
cause your adkats was spamming my server and I sent you money for you to stop it ?

**Comment by danfango** [comment]
and you blamed rconnet lol

**Comment by ColColonCleaner** [comment]
@danfango Issue #910 was to disable instance checks and allowed compatibility with rconnet's chat logs. They kept chat logs of hidden admin say messages for some reason. This is how software development works when you interact with other systems. They sometimes don't mesh. I would assume after 37 years in IT that would be a given for you, and would yield more than 35K per year.

Stat logger crashing your procon instance is an anomaly that needs to be investigated, not evidence of a wider issue. Likely a compatibility problem with something on your machines/layers. The claim of thousands of users being stable is used to show there isn't a grass roots problem with the system, and it's only a compatibility issue or a configuration issue with your system.

**Comment by danfango** [comment]
what a fucking douchebag

**Comment by danfango** [comment]
Your the fucker thats writing plugind not me

**Comment by danfango** [comment]
really your and thord party plugins crash a clean procon ?

**Comment by ColColonCleaner** [comment]
@danfango You really have a vendetta don't you. Two systems did not work together, AdKats and Rconnet chat logs, and you wanted them to work together. I told you what would be needed for them to work together and you paid for that patch. It's very simple, and makes perfect sense. Calm down.

**Comment by danfango** [comment]
you reall want me to test your incompetencancy across al platforms ?

**Comment by danfango** [comment]
im not gonna cause i have a life

**Comment by danfango** [comment]
stats logger crashers procon ok

**Comment by danfango** [comment]
simple fucking accept it

**Comment by danfango** [comment]
writtewn by a nobody and no longer supported

**Comment by danfango** [comment]
I rest my case

**Comment by danfango** [comment]
at least when you write a code for a parent program you might test it doesn't crash the parent program

**Comment by danfango** [comment]
but hey procon is free and it's plugins are written buy idiots (amatuers) so when it crashes should we really care ?... It does some things well and does others badly maybe we should be thankful

**Comment by danfango** [comment]
good luck to get adkats running on 2 unsupported plugind that crash procon ???? but hey hold on there are 1000's pf happy users ? oh well

**Comment by danfango** [comment]
there on windows xp ?

**Comment by ColColonCleaner** [comment]
@danfango Resting a case without substance doesn't mean anything. Yes, free software likely contains bugs, and it looks like you ran into one which has now become a log in your eye and you can't see past it. Not everyone who contributes to a community and releases code for free is a professional, so provide the information they need to solve your issue on your platform.

This is how community driven programming works. Find an issue with a free script and if someone is willing to fix it, ask them or pay them. We all have lives, this is not our world. It's a place we can contribute together and make the admin-ship of battlefield as a whole better, and right now you're polluting it instead of helping solve the issue.

**Comment by danfango** [comment]
Simple thing is your plugins work or they don't there is no massive mystery to procon or it's plugins ... just shit programmers that don't test or give a shit about their plugins

**Comment by danfango** [comment]
@ColColonCleaner you tell us under what circumstances/opewrating systems can we get this  to work

**Comment by danfango** [comment]
cause surely if you are writing code u casn tell us under what conditions that code will work ?

**Comment by ColColonCleaner** [comment]
It's been ran under windows 7, windows 2012 R2, windows 10, and some versions of mono linux. I would suggest staying away from linux though since there is an issue connecting to github over HTTPS from linux with the .net 3.5 libraries.

**Comment by danfango** [comment]
anyway good thank god I'm closing my servers and dont need to worry about procon or rconnet

**Comment by ColColonCleaner** [comment]
And we'll work on fixing @Semperfi-mantis 's specific issue. I hope your time away from battlefield helps, along with the therapy and AA meetings of course. Cheers.

**Comment by Semperfi-mantis** [comment]
@ColColonCleaner  Thx mate, ill wait.. adkats is running fine have stats logger off now.
I know Stats logger is causing the crash of procon.. and on a empty server, or with me only in it no crash.. but if i turn it on, on a half/full server it crashes( but the thing is i get the message procon is crahsed, but its still running) if i turn stats logger of, all is just fine!

**Comment by ColColonCleaner** [comment]
Send me a PM on the myrcon forums with credentials to access your layer/plugins and I'll take a look at it. Please open a new issue for this so I can track it without the noise above.

**Comment by ColColonCleaner** [comment]
Adding reference to #872 since that appears to be nearly the same issue but not really. The one posted by @Semperfi-mantis has multiple quotes instead of just a single pair.

**Comment by danfango** [comment]
Hes not running it on a layer he is running it on his local pc same as me maybe your 3000 users are using layers and that's why it's working

On 28 Jun 2017 22:45, Daniel Gradinjan <notifications@github.com> wrote:

Adding reference to #872<https://github.com/AdKats/AdKats/issues/872> since that appears to be nearly the same issue but not really. The one posted by @Semperfi-mantis<https://github.com/semperfi-mantis> has multiple quotes instead of just a single pair.

**Comment by Prophet731** [comment]
@ColColonCleaner I don't remember but I do know that what's its doing it's just converting the IP to an integer instead of a string. It crashing is probably because it's expecting an IP address but was getting a blank string. Easy fix to check if it's empty or not.

**Comment by Semperfi-mantis** [comment]
@ColColonCleaner  accept Blackmantis on myrcon thats me

**Comment by ColColonCleaner** [comment]
@danfango Enough, let it go. Some groups with 15+ instances of procon on the same local machine or VPS run without issues, and I run local instances of procon for testing on full servers constantly. All a "layer" means is an instance of procon that other instances connect to, and runs without the UI thread. I truly hope my fellow software engineers do not end up like you in their later years.

@Prophet731 Alright i'll run some queries on his DB to check what values are invalid and why.

@Semperfi-mantis I responded to your message.

**Comment by Prophet731** [comment]
It won't be on  the database. It will be with the query it's trying to do. Remember when I said I wanted to store the ip addresses in numeric format for easier searching when we were talking about that economy service?

**Comment by ColColonCleaner** [comment]
@Prophet731 If there is an invalid value being pushed by AdKats, pulled by stat logger, then updated back to the database through that inet function it would fail. That was just the case I was worried about. But it could be code in the stat logger plugin itself.

**Comment by Prophet731** [comment]
More than likely in stat logger but i'm confused as to why now this is an issue. I mean we are probably the ones who ran that plugin more than everyone but we've never seen that issue.

**Comment by Prophet731** [comment]
@ColColonCleaner 

I've created a git project for the stats logger so we can keep track of changes made to the file more easily. I might wanna take up the task of maybe fixing the flaws in the plugin that we've known about. I'll add you as a collab so you could write to it.

https://github.com/Prophet731/battlefield-stats-logger-procon

**Comment by Prophet731** [comment]
@Semperfi-mantis 

I've modified the stat logger plugin to check if the ip valid for it's conversion. Can you try it and see if that fixes the problem? You can download [here](https://github.com/Prophet731/battlefield-stats-logger-procon/archive/develop.zip)

**Comment by Semperfi-mantis** [comment]
ok, i will try it

**Comment by Semperfi-mantis** [comment]
Till now it keeps running, normal when i set stats logger on, procon crashes. now i have all stats logger on, on 5 servers no crash yet :D

**Comment by danfango** [comment]
FFS you lot still at it ?

On 1 Jul 2017 17:09, Semperfi-mantis <notifications@github.com> wrote:

ok, i will try it

**Comment by ColColonCleaner** [comment]
@danfango I wonder what kind of IT you're in considering you don't seem to have any basic idea how troubleshooting a problem works. Of course we're still at it, the problem still exists.

**Comment by ColColonCleaner** [comment]
@Prophet731 Thanks for adding that patch. If @Semperfi-mantis still has a positive result after a while I'd like to merge your version with the unofficial one that tyger posted, and we can take over that stat logger thread and make this the new official version.

**Comment by Semperfi-mantis** [comment]
damm i unsubscribed on the post clicked the wrong link in email lol..

I will keep posting tho if it crashed, till now still running :D

**Comment by danfango** [comment]
How can you run adkats without stats logger it's a dependancy

On 1 Jul 2017 18:41, Semperfi-mantis <notifications@github.com> wrote:

damm i unsubscribed on the post clicked the wrong link in email lol..

I will keep posting tho if it crashed, till now still running :D

**Comment by ColColonCleaner** [comment]
@danfango Technically you can. The only reason AdKats requires stat logger as a dependency is because that's what the database structure is based off of. Stat logger was the most popular database tool used by admin groups so AdKats was built around that structure initially. If you don't care at all about player stats, after stat logger has been ran for one round (enough that AdKats fully starts up), then you can run AdKats without the stat logger plugin running.

**Comment by danfango** [comment]
Anyway no one cares

On 1 Jul 2017 22:19, Daniel Gradinjan <notifications@github.com> wrote:

@danfango<https://github.com/danfango> Technically you can. The only reason AdKats requires stat logger as a dependency is because that's what the database structure is based off of. Stat logger was the most popular database tool used by admin groups so AdKats was built around that structure initially. If you don't care at all about player stats, after stat logger has been ran for one round (enough that AdKats fully starts up), then you can run AdKats without the stat logger plugin running.

**Comment by danfango** [comment]
Really all this hacking and shit and you care about stats logger ?

On 1 Jul 2017 22:19, Daniel Gradinjan <notifications@github.com> wrote:

@danfango<https://github.com/danfango> Technically you can. The only reason AdKats requires stat logger as a dependency is because that's what the database structure is based off of. Stat logger was the most popular database tool used by admin groups so AdKats was built around that structure initially. If you don't care at all about player stats, after stat logger has been ran for one round (enough that AdKats fully starts up), then you can run AdKats without the stat logger plugin running.

**Comment by ColColonCleaner** [comment]
@danfango Of course. Taking advantage of historical player data is very useful to the admins using AdKats. It makes that data available via in-game commands and keeps all the information in one place. A lot of people already used stat logger for a couple years before AdKats came on the scene, so being able to capitalize on that meant a lot to people. No need to reset their player data.

There is also no reason for me to stop responding seriously to these drunken ramblings. Perhaps at some point the reality of things will set in on your side.

**Comment by Prophet731** [comment]
@ColColonCleaner 

My modified version actually incorps tygers changes so mine would be the latest unofficial version.

**Comment by ColColonCleaner** [comment]
@Prophet731 Perfect. I have a 4 day weekend for the holiday so I'll make some moves on the rcon forums and get that thread updated with the two of us as the devs.

**Comment by Prophet731** [comment]
@ColColonCleaner 

Alright, i'll be heading home on tuesday for a few days so i'll have my free time then too.

**Comment by Prophet731** [comment]
@ColColonCleaner 

Also i'll purchase a bf3, bf4, & bfh server so that we can have at least something to test against. All together it'll be under $100 for 3 months each. Lowest player count obviously.

**Comment by ColColonCleaner** [comment]
@Prophet731 Maybe don't do that just yet. If someone has an issue and needs it looked into we can always request access to that server.
